tests/lean/run/fin_coercions.lean

```diff
@@ -0,0 +1 @@
+example (x : Fin (n + 1)) (h : x < n) : Fin (n + 1) := x.succ.castLT (by simp [h])
```

Comment:
Could you add some tests like
```
/-- ... err msg ... -/
#guard_msgs in #check fun (n : Nat) => (n : Fin 3)

section
attribute [local instance] Fin.instNatCast
variable (m : Nat) (n : Fin 3)
/-- info: n < ↑m : Prop -/
#guard_msgs in #check n < m
end
```
to make sure it doesn't get added back accidentally?

(I feel like I really ought to have added some back when I noticed this issue! Sorry)